### PR TITLE
Storage test enhancements

### PR DESCRIPTION
**Pull request by Jinming-Hu**
closes https://github.com/Azure/azure-sdk-for-cpp/issues/614

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Comment by Jinming-Hu** [comment]
@katmsft I'd like to know this [this](https://github.com/Azure/azure-sdk-for-cpp/blob/07c5cba59f63eb526eae7a5de9770e9216749bd8/sdk/storage/azure-storage-files-shares/test/share_service_client_test.cpp#L247) test case was disabled. I didn't find any related issue.

**Comment by Jinming-Hu** [comment]
@katmsft also, let's add LAT back and see if it works now

**Review by vhvb1989 — APPROVED** [review]

File: sdk/storage/azure-storage-blobs/test/blob_service_client_test.cpp [hunk 1]
```diff
@@ -338,10 +338,6 @@ namespace Azure { namespace Storage { namespace Test {
         = Blobs::BlobServiceClient(InferSecondaryUrl(m_blobServiceClient.GetUrl()), keyCredential);
     auto serviceStatistics = *secondaryServiceClient.GetStatistics();
     EXPECT_FALSE(serviceStatistics.GeoReplication.Status.Get().empty());
-    if (serviceStatistics.GeoReplication.LastSyncedOn.HasValue())
-    {
-      EXPECT_TRUE(IsValidTime(serviceStatistics.GeoReplication.LastSyncedOn.GetValue()));
-    }
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/test/blob_service_client_test.cpp:344]: Why are we removing this?

**Comment by ahsonkhan** [comment]
Looks like these uncommented tests are still failing.

Refreshing the run since the build is gone:
image

/azp run cpp - storage

**Comment by RickWinter** [comment]
@Jinming-Hu This PR is nearing 30 days old.  If this PR is abandoned please close it.  If the PR is still valid please work to get it submitted ASAP.

**Comment by ahsonkhan** [comment]
Moving the milestone to March since this doesn't seem ready for the Feb release.

**Comment by ahsonkhan** [comment]
This PR is now really old with no recent activity. Please feel free to re-open when you plan to work on it.

**Pull request closed without merging**